charms/argo-controller/src/charm.py

```diff
@@ -54,260 +80,93 @@ def __init__(self, *args):
-                                        "update",
-                                        "patch",
-                                        "delete",
-                                    ],
-                                },
-                                {
-                                    "apiGroups": ["argoproj.io"],
-                                    "resources": [
-                                        "workflowtaskresults",
-                                    ],
-                                    "verbs": [
-                                        "list",
-                                        "watch",
-                                        "delete",
-                                    ],
-                                },
-                                {
-                                    "apiGroups": [""],
-                                    "resources": ["events"],
-                                    "verbs": ["create", "patch"],
-                                },
-                                {
-                                    "apiGroups": ["policy"],
-                                    "resources": ["poddisruptionbudgets"],
-                                    "verbs": ["create", "get", "delete"],
-                                },
-                                {
-                                    "apiGroups": ["coordination.k8s.io"],
-                                    "resources": ["leases"],
-                                    "verbs": ["create", "get", "update"],
-                                },
-                                {
-                                    "apiGroups": [""],
-                                    "resources": ["secrets"],
-                                    "verbs": ["get"],
-                                },
-                            ],
-                        }
-                    ],
+        self.kubernetes_resources = self.charm_reconciler.add(
+            component=KubernetesComponent(
+                charm=self,
+                name="kubernetes:auth-crds-cm-and-secrets",
+                resource_templates=K8S_RESOURCE_FILES,
+                krh_resource_types={
+                    ClusterRole,
+                    ClusterRoleBinding,
+                    ConfigMap,
+                    CustomResourceDefinition,
+                    Role,
+                    RoleBinding,
+                    Secret,
+                    ServiceAccount,
                 },
-                "service": {
-                    "updateStrategy": {
-                        "type": "RollingUpdate",
-                        "rollingUpdate": {"maxUnavailable": 1},
-                    },
+                krh_labels=create_charm_default_labels(
+                    self.app.name,
+                    self.model.name,
+                    scope="auth-crds-cm-and-secrets",
+                ),
+                context_callable=lambda: {
```

Comment:
This inline JSON object makes it a bit hard to read. Is it possible to create `_generate_context()` helper that would return this object and use here it instead?
Not sure if it is possible. Just nitpicking.

Reply:
Updated